src/odemis/driver/actuator.py

```diff
@@ -940,6 +940,203 @@ def reference(self, axes):
         return f
 
 
+class LinearActuator(model.Actuator):
+    """
+    A generic actuator component which allows moving on a linear axis. It is actually a
+    wrapper to just one axis/actuator and it can also apply cyclic moves e.g. in case the
+    actuator moves a filter wheel.
+    """
+
+    def __init__(self, name, role, children, axis_name, offset,
+                 moves_before_ref=10, inverted=None, **kwargs):
+        """
+        name (string)
+        role (string)
+        children (dict str -> actuator): axis name (in this actuator) -> actuator to be used for this axis
+        axis_name (str): axis name in the child actuator
+        offset (float): axis offset (negative of reference position value)
+        moves_before_ref (int): number of moves before referencing axis
+        positions (set or dict value -> str): positions where the actuator is allowed to move
+        cycle (float): if not None, it means the actuator does a cyclic move and this value represents a full cycle
+        """
+        if inverted:
+            raise ValueError("Axes shouldn't be inverted")
+
+        if len(children) != 1:
+            raise ValueError("FixedPositionsActuator needs precisely one child")
+
+        self._move_sum = 0
+        self._position = {}
+        self._referenced = {}
+        axis, child = children.items()[0]
+        self._axis = axis
+        self._child = child
+        self._caxis = axis_name
+        self._offset = offset
+
+        self._moves_before_ref = moves_before_ref  # max number of moves before referencing
```

Comment:
You could add, relatively easily, support to never force the referencing. If it's `None`, then referencing should never be automatically done. Just store None, and check with `if self._moves_before_ref is not None` before referencing.